src/lib.rs

```diff
@@ -954,6 +999,55 @@ unsafe extern "C" fn settings_enable_modified(
     true
 }
 
+#[cfg(feature = "auto-splitting")]
+unsafe extern "C" fn settings_file_select_modified(
+    data: *mut c_void,
+    _props: *mut obs_properties_t,
+    _prop: *mut obs_property_t,
+    settings: *mut obs_data_t,
+) -> bool {
+    let default_setting_string = CStr::from_ptr(DEFAULT_AUTO_SPLITTER_LIST_SETTING);
+    let list_setting_string = CStr::from_ptr(obs_data_get_string(
+        settings,
+        SETTINGS_AUTO_SPLITTER_SETTINGS_LIST,
+    ));
+
+    if list_setting_string == default_setting_string {
+        return false;
+    }
+
+    let state: &mut State = &mut *data.cast();
+
+    let value = obs_data_get_string(settings, SETTINGS_AUTO_SPLITTER_SETTINGS_FILE_SELECT);
+
+    obs_data_set_string(settings, SETTINGS_AUTO_SPLITTER_SETTINGS_FILE_SELECT, value);
```

Comment:
What does this do? It looks like it sets back the value that we just read?

Reply:
I don't really know. I just saw a similar line in the boolean / checkbox case and copied it. I can delete it and re-test stuff.

Reply:
I have a branch that changes that here: https://github.com/AlexKnauth/obs-livesplit-one/tree/file-select-test

At some point in the next day or so I'll find time to test it.

Reply:
In the `Remove redundant set=get` I removed this line in both the file-select case and the bool case. I tested it with a contrived auto-splitter that put the "wrong" type in, and it seemed to work just fine, as long as this change goes together with the `Remove Some(_) case, merge with None case` commit.